Tasks/JenkinsDownloadArtifacts/ArtifactDetails/JenkinsRestClient.ts

```diff
@@ -141,4 +212,127 @@ export class JenkinsRestClient {
+                }
+                else {
+                    console.log(tl.loc("FoundJenkinsJobDetails", jobName, buildId, branchName));
+                    defer.resolve(new JenkinsJobDetails(jobName, buildId, jobType, branchName));
+                }
+            });
+
+            return defer.promise;
+        }
+    }
+
+    public GetLastSuccessfulBuild(): Q.Promise<JenkinsJobDetails> {
+        let defer = Q.defer<JenkinsJobDetails>();
+        let jobType: string = "";
+        const jobName: string = tl.getInput('jobName', true);
+
+        this.GetJobType().then((result: string) => {
+            jobType = result;
+        }).fin(() => {
+
+            let jenkinsTreeParameter = "lastSuccessfulBuild[id,displayname]";
+            let handlerbarSource = "{{lastSuccessfulBuild.id}}";
+            const isMultibranchPipeline = jobType.toLowerCase() === JenkinsJobTypes.MultibranchPipeline.toLowerCase();
+
+            tl.debug(`Found Jenkins Job type ${jobType} and isMultibranchPipeline ${isMultibranchPipeline}`);
+            if (isMultibranchPipeline) {
+                jenkinsTreeParameter = "jobs[name,lastSuccessfulBuild[id,displayName,timestamp]]"
+                handlerbarSource = "{{#with (selectMaxOf jobs 'lastSuccessfulBuild.timestamp') as |job|}}{ \"branchName\": \"{{job.name}}\", \"buildId\": \"{{job.lastSuccessfulBuild.id}}\" }{{/with}}"
+            }
+
+            const lastSuccessfulUrlSuffix: string = `/api/json?tree=${jenkinsTreeParameter}`;
+            this.DownloadJsonContent(lastSuccessfulUrlSuffix, handlerbarSource, null).then((result) => {
+                let buildId: number;
+                let branchName: string;
+                let succeeded: boolean = false;
+
+                if (isMultibranchPipeline) {
+                    try {
+                        let jsonResult = JSON.parse(result);
+                        branchName = jsonResult["branchName"];
+                        buildId = parseInt(jsonResult["buildId"]);
+                        tl.debug(`Found branchName: ${branchName}, buildId: ${buildId}`);
+                    } catch(error) {
+                        defer.reject(new Error(tl.loc("CouldNotGetLastSuccessfulBuildNumber", error)));
+                    }
+                }
+                else {
+                    buildId = parseInt(result);
+                }
+
+                if (isNaN(buildId)) {
+                    defer.reject(new Error(tl.loc("InvalidBuildId", buildId)));
+                }
+                else {
+                    defer.resolve(new JenkinsJobDetails(jobName, buildId, jobType, branchName));
+                }
+
+            }, (error) => {
+                defer.reject(new Error(tl.loc("CouldNotGetLastSuccessfulBuildNumber", error)));
+            });
+
+        });
+
+        tl.debug(tl.loc('GetArtifactsFromLastSuccessfulBuild', jobName));
+
+        return defer.promise;
+    }
+
+    public static JenkinsFolderSeperator: string = "/";
```

Comment:
there isn't any folder so far... should we name constant as BranchPathSeparator?

Reply:
Done